### PR TITLE
scheduler: preserve output ids when merging batches

### PR DESCRIPTION
**Pull request by wuyingjun-lucky**
## Summary
- preserve pending output ids when merging batches across prefill/decode boundaries where one side has already moved token ids into `input_ids`
- add focused unit coverage for the `output_ids is None` merge cases
- inline the `sgl_kernel` stub in the new unit test so it can run without pulling in heavy `test_utils` dependencies

## Testing
- `PYTHONPATH=/home/wuyingjun/llm/sglang-git/python:/home/wuyingjun/llm/sglang-git:/tmp/sglang-testdeps python -m pytest /home/wuyingjun/llm/sglang-git/test/registered/unit/managers/test_schedule_batch_merge.py /home/wuyingjun/llm/sglang-git/test/registered/rl/test_pause_generation_tensor_consistency.py`

### CODE REVIEW

File: python/sglang/srt/managers/schedule_batch.py [hunk 1]
```diff
@@ -2395,7 +2395,14 @@ def merge_batch(self, other: "ScheduleBatch"):
         self.orig_seq_lens = torch.cat([self.orig_seq_lens, other.orig_seq_lens])
         self.out_cache_loc = None
         self.seq_lens_sum += other.seq_lens_sum
-        if self.output_ids is not None:
+        # Batches can be merged across prefill/decode boundaries. A prepared
+        # decode batch has already moved its pending token ids into
+        # ``input_ids`` and cleared ``output_ids``, while prebuilt decode
+        # batches still populate ``output_ids``. Preserve whichever side still
+        # has pending token ids and concatenate only when both do.
+        if self.output_ids is None:
```
Comment (hnyls2002) [inline python/sglang/srt/managers/schedule_batch.py:2403]: This can not be None, if it is None, then there must be something wrong on the other side.

**Review by hnyls2002 — CHANGES_REQUESTED** [review]

**Comment by hnyls2002** [comment]
When using AI to fix bugs, please provide at least some context or a reproduction. Only a few lines are not persuasive
